src/TMS/dev/TMP117.cpp

```diff
@@ -0,0 +1,29 @@
+//
+// Created by Ishan on 1/23/2023.
+//
+
+#include <TMS/dev/TMP117.hpp>
+#include <EVT/io/I2C.hpp>
+
+namespace TMS {
+
+TMP117::TMP117(IO::I2C& i2c, uint8_t deviceID): deviceID(deviceID), i2c(i2c) { }
+
+uint16_t TMP117::readTemp() {
+    uint16_t input;
+    uint8_t tempBytes[2];
+    uint16_t byte1;
+
+    i2c.readReg(deviceID, tempReg, regLength, tempBytes, valueLength);
```

Comment:
```suggestion
    i2c.readReg(deviceID, &tempReg, regLength, tempBytes, valueLength);
```
When you change `tempReg` to a `uint8_t`, this will need get its address with `&`